dynawo/sources/Models/CPP/Controls/Voltage/ModelVoltageMeasurementsUtilities/DYNModelVoltageMeasurementsUtilities.cpp

```diff
@@ -0,0 +1,442 @@
+//
+// Copyright (c) 2022, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+// This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools
+// for power systems.
+//
+
+/**
+ * @file  DYNModelVoltageMeasurementsUtilities.cpp
+ *
+ * @brief Measurements utility model
+ *
+ * Model to compute some aggregations of the voltages connected to the (sub)network.
+ * Only running models are taken into account for the computations.
+ *
+ * Aggregations implemented:
+ * -> Minimum
+ * -> Maximum
+ * -> Average
+ *
+ */
+
+#include <sstream>
+#include <vector>
+#include "PARParametersSet.h"
+
+#include "DYNModelVoltageMeasurementsUtilities.h"
+#include "DYNModelVoltageMeasurementsUtilities.hpp"
+#include "DYNModelConstants.h"
+#include "DYNElement.h"
+#include "DYNMacrosMessage.h"
+#include "DYNElement.h"
+#include "DYNCommonModeler.h"
+#include "DYNTrace.h"
+#include "DYNVariableForModel.h"
+#include "DYNParameter.h"
+
+using std::vector;
+using std::string;
+using std::map;
+using std::stringstream;
+
+using boost::shared_ptr;
+
+using parameters::ParametersSet;
+
+/**
+ * @brief ModelVoltageMeasurementsUtilitiesFactory getter
+ *
+ * @return A pointer to a new instance of ModelVoltageMeasurementsUtilitiesFactory
+ */
+extern "C" DYN::SubModelFactory* getFactory() {
+  return (new DYN::ModelVoltageMeasurementsUtilitiesFactory());
+}
+
+/**
+ * @brief ModelVoltageMeasurementsUtilitiesFactory destroy method
+ */
+extern "C" void deleteFactory(DYN::SubModelFactory* factory) {
+  delete factory;
+}
+
+/**
+ * @brief ModelVoltageMeasurementsUtilities getter
+ *
+ * @return A pointer to a new instance of ModelVoltageMeasurementsUtilities
+ */
+extern "C" DYN::SubModel* DYN::ModelVoltageMeasurementsUtilitiesFactory::create() const {
+  DYN::SubModel* model(new DYN::ModelVoltageMeasurementsUtilities());
+  return model;
+}
+
+/**
+ * @brief ModelVoltageMeasurementsUtilitiesFactory destroy method
+ */
+extern "C" void DYN::ModelVoltageMeasurementsUtilitiesFactory::destroy(DYN::SubModel* model) const {
+  delete model;
+}
+
+namespace DYN {
+
+/**
+ * @brief ModelVoltageMeasurementsUtilities model default constructor
+ *
+ */
+ModelVoltageMeasurementsUtilities::ModelVoltageMeasurementsUtilities() :
+ModelCPP("voltageMeasurementsUtilities"),
+nbConnectedInputs_(0), step_(12.), isInitialized_(false) {
```

Comment:
This came from a discussion in which the default value for this would be around 10s. But maybe, as you mention, we can make giving this parameter mandatory and set some other value here (0 being a possibility). It all depends on the usual code quality / rules